visual-tests/imports/helpers.js

```diff
@@ -0,0 +1,71 @@
+export const helpers = {
+  defaultHOTVersion: '12.2.0',
+  defaultHOTWrapper: 'js',
+
+  screenshotsDirectory: './tests/screenshots',
+  screenshotsExtension: 'png',
+
+  selectors: {
+    mainTable: 'div.handsontable.htRowHeaders.htColumnHeaders',
+    mainTableBody: '> .ht_master.handsontable table tbody',
+    mainTableHead: '> .ht_clone_top.handsontable table thead',
+    mainTableFirstColumn: '> .ht_clone_inline_start.ht_clone_left.handsontable table tbody',
```

Comment:
`mainTable points to the main table container`
Sure, I changed the class of `mainTable` already, but rest of points was done intentionally. Correct me if I mistaken something  ;) 

`mainTableBody points to the tbody element on the master overlay`:

that's true - it's because we are looking for cells and their content in `master table`, they are working correctly, are available for the testing tool and any weird solution was not needed, but here:

`mainTableHead points to the thead element, but on the top overlay (which makes sense if we're targeting the topmost elements)`

I did it in this way, cause in `thead` we are looking for buttons or columns headers which are not available for testing tool if you look for them in `master table` - e.g. if you want to click `button.changeType`, testing tool will return error `element not visible`, cause it's overlayed by `clone_top table` - question is: if this clone always exists? If no, then there is other way to workaround it, but requires one more thing from tester side, or I can rebuild helpers to detect if clone exists.

`mainTableFirstColumn points the the tbody element (so maybe the "column" in the property name can be misleading?) on the left overlay, which might not work if there are no frozen columns`

I wanted to give tester easy access to the left column without any unnecessary moves from their side and from tester point of view this returns indeed left column of main table - they can easy search for given cell in it. But not sure about this part with `frozen columns` - would you tell me more?

Reply:
In general, the additional overlays are used for:
- [Row headers](https://handsontable.com/docs/javascript-data-grid/row-header/) and [column headers](https://handsontable.com/docs/javascript-data-grid/column-header/)
- Fixed/frozen [rows](https://handsontable.com/docs/javascript-data-grid/row-freezing/) (also on the [bottom of the table](https://handsontable.com/docs/javascript-data-grid/api/options/#fixedrowsbottom)) and [columns](https://handsontable.com/docs/javascript-data-grid/column-freezing/)
- The "corner" top-left cell on the intersection of row and column headers

If you're trying to target the _topmost_ DOM element representing the row/column headers, it'll always be attached to the `top` overlay (column header) / `left/start` overlay (row header).
If you're trying to target the _topmost_ DOM element representing a _cell_ things get more complicated. It can be stored under either the `master` overlay or the `top`/`left/start` overlay, depending on whether they fall into the fixed/frozen rows/columns range.

For example:
```js
const hot = new Handsontable(container, {
  // (...)
  fixedRowsTop: 2,
  fixedColumnsStart: 2,
});

// Cells (0, 0), (1, 1), (0, 5), (5, 1) etc have their duplicates on their own respective additional overlays
// Cells (2, 2), (5, 5), (3, 5), (5, 3) etc don't
```

Reply:
well, I had said that I changed this selector `mainTable: 'div.handsontable.htRowHeaders.htColumnHeaders',` already just to `mainTable: 'div.handsontable' and it was true, but after that I tested it and testing tool has problems, cause in pages with examples there is more than one element with this class (and probably that was reason why I did it in current way at the same beginning).

I tried to find just this `.handsontable` element in root or somewhere in common place on page, but unfortunatelly each of our examples has different DOM structure and it is not possible to do that in this way. We can do it somehow by looking for elements by `querySelectorAll` for example, but I'm not sure if it won't bring problems in future.

My proposition is just to leave it in current form for a while -> unify examples DOM in next step -> change this selector in helpers without touching tests itself.

What do you think?

Reply:
> in pages with examples there is more than one element with this class

`div.handsontable.htRowHeaders.htColumnHeaders` could easily return more elements as well, either on a page with multiple Handsontable instances or on an instance with Handsontable-based editors being open.

I don't think we should make the testing tool work only with our current, single test case, but rather make it as generic as possible to make it as future-proof as we possibly can.

Generally speaking, using CSS classes to target one, specific DOM element is not the best approach. Maybe adding `id`s to the Handsontable instances would be a better idea? For example, we could assume each test case would contain Handsontable instances `id`'d as `#example1`, `#example2`, `#example3` etc - with `#example1` being the main, default one?